google/cloud/bigquery/client.py

```diff
@@ -3680,7 +3680,7 @@ def insert_rows(
         if selected_fields is not None:
             schema = selected_fields
 
-        if len(schema) == 0:
+        if not schema:
```

Comment:
I'm curious, could you make `len(schema_object)` work? I guess this is to allow `None` as `schema` now, too? If `table.schema` could indeed return `None` now, that might be considered a breaking change, depending on how strict we want to be about types.

Either way, folks might have to change their code to deal with missing schema, so something to consider. Maybe we return an object of `len(...) == 0` for now and have a warning that it will be changed to `None` in future?

Reply:
I was under the impression that `Table.schema` can currently be set to `None`. It appears that we allow for that in the existing code:

https://github.com/googleapis/python-bigquery/blob/d4d39acb8574f0d06d4e490b859e5fe6b57d0d9e/google/cloud/bigquery/table.py#L458C1-L466C84

```python
@schema.setter
def schema(self, value):
    api_field = self._PROPERTY_TO_API_FIELD["schema"]
    if value is None:
        self._properties[api_field] = None
    else:
        value = _to_schema_fields(value)
        self._properties[api_field] = {"fields": _build_schema_resource(value)}
```

Reply:
I switched it back to the old check.